execute/plugin_functions.go

```diff
@@ -227,15 +269,47 @@ func truncateObservation(
 	// If the encoded observation is too large, start filtering data.
```

Comment:
Should costly messages be removed first?

Reply:
going to do it in next PR